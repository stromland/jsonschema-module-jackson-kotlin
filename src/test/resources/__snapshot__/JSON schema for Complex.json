{
  "$schema" : "https://json-schema.org/draft/2020-12/schema",
  "type" : "object",
  "properties" : {
    "configs" : {
      "default" : [ {
        "type" : "DEVELOPMENT",
        "enabled" : true,
        "properties" : { },
        "test" : { }
      } ],
      "type" : "array",
      "items" : {
        "type" : "object",
        "properties" : {
          "enabled" : {
            "type" : "boolean",
            "default" : true
          },
          "properties" : {
            "type" : "object",
            "additionalProperties" : {
              "type" : "string"
            },
            "default" : { }
          },
          "test" : {
            "type" : "object",
            "additionalProperties" : {
              "type" : "string"
            }
          },
          "type" : {
            "type" : "string",
            "enum" : [ "DEVELOPMENT", "PRODUCTION" ]
          }
        },
        "required" : [ "test", "type" ],
        "default" : [ {
          "type" : "DEVELOPMENT",
          "enabled" : true,
          "properties" : { },
          "test" : { }
        } ]
      }
    },
    "name" : {
      "type" : "string"
    },
    "version" : {
      "type" : [ "string", "null" ]
    }
  },
  "required" : [ "name" ]
}