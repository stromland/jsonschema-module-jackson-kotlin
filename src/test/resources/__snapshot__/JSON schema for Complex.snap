{
  "$schema" : "http://json-schema.org/draft-07/schema#",
  "definitions" : {
    "ComplexConfig" : {
      "type" : "object",
      "properties" : {
        "enabled" : {
          "type" : "boolean",
          "default" : true
        },
        "properties" : {
          "allOf" : [ {
            "$ref" : "#/definitions/Map(String,String)"
          }, {
            "default" : { }
          } ]
        },
        "type" : {
          "$ref" : "#/definitions/ConfigType"
        }
      },
      "required" : [ "type" ]
    },
    "ConfigType" : {
      "type" : "string",
      "enum" : [ "DEVELOPMENT", "PRODUCTION" ]
    },
    "Map(String,String)" : {
      "type" : "object"
    }
  },
  "type" : "object",
  "properties" : {
    "configs" : {
      "default" : [ {
        "type" : "DEVELOPMENT",
        "enabled" : true,
        "properties" : { }
      } ],
      "type" : "array",
      "items" : {
        "$ref" : "#/definitions/ComplexConfig"
      }
    },
    "name" : {
      "type" : "string"
    },
    "version" : {
      "type" : [ "string", "null" ]
    }
  },
  "required" : [ "name" ]
}